spec/factories/project_steps.rb

```diff
@@ -0,0 +1,49 @@
+FactoryGirl.define do
+  factory :project_step do
+    # project_table 'Loan'
+    # project_id { create(:loan).id }
+    project { create(:loan) }
+    person
+    # need to make sure parent saved before assigning these
+    # summary { Faker::Lorem.sentence }
+    # details { Faker::Lorem.paragraph }
+    scheduled_date { Faker::Date.between(Date.civil(2004, 01, 01), Date.today) }
+    completed_date { Faker::Date.between(Date.civil(2004, 01, 01), Date.today) }
+    is_finalized [true,false].sample  ##JE todo: is ther a better pattern to assign a boolean?
+    type_option_id { ProjectStep::TYPE_OPTIONS.values.sample }
+
+    trait :completed do
+      completed_date { Faker::Date.between(date, Date.today) }
+    end
+
+    trait :past do
+      scheduled_date { Faker::Date.backward }
+    end
+
+    trait :future do
+      scheduled_date { Faker::Date.forward }
+    end
+
+    # is this even valid?
+    # trait :for_loan do
+    #   transient { loan_id 0 }
+    #   after(:build) do |event, evaluator|
```

Comment:
Believe it or not, this did work. It doesn't really have to work anymore though, it was built in to save space & energy on the calling side not having to manually set the correct table and stuff manually. The `for_loan: loan.id` calls are probably just going to need to be replaced, wherever they show up, with `project: loan` calls now that it's a proper association and this stuff can be deleted. 

If we wanted to keep the `for_loan` syntax though, it wouldn't look that much different:

```ruby
transient { loan_id 0 }
after(:build) do |step, evaluator|
  step.project = Loan.find(loan_id)
end
```